ddl/table.go

```diff
@@ -96,8 +97,16 @@ func onCreateView(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error)
 		// none -> public
 		tbInfo.State = model.StatePublic
 		tbInfo.UpdateTS = t.StartTS
-		err = t.CreateTable(schemaID, tbInfo)
+		if oldTbInfoID > 0 && orReplace {
+			err = t.DropTable(schemaID, oldTbInfoID, true)
+			if err != nil {
+				job.State = model.JobStateCancelled
+				return ver, errors.Trace(err)
+			}
+		}
+		err = t.CreateTableOrView(schemaID, tbInfo)
 		if err != nil {
+			job.State = model.JobStateCancelled
```

Comment:
Remove this, `model.JobStateCancelled` is dangerous. Return an error directly, and then DDL will retry.